composer/datasets/in_context_learning_evaluation.py

```diff
@@ -1159,78 +890,97 @@ def build_icl_dataloader(
     prompt_string: str,  # e.g. 'translate english to french:'
     example_delimiter: str,  # e.g. '\n'
     continuation_delimiter: str,  # e.g. ''
+    hf_loading_vars: dict,
+    hf_parsing_vars: dict,
     destination_path: str,
-    question_prelimiter: str = '',  # e.g. 'Question: '
-    cot_delimiter: str = '',
-    fewshot_random_seed: int = 1234,
-    pass_at_k: int = 1,
-    generations_per_sample: int = 1,
+    prelimiter: str,  # e.g. 'Question: '
```

Comment:
so, changing the name of this param is a breaking change. if you are really attached we can deprecate the old one and then remove in a couple versions.

Reply:
I'm not. question_prelimiter and code_prelimiter were doing the same thing in code tasks and QA tasks, so I figured it'd be best to combine them into a more general parameter. Can revert if you'd prefer

Reply:
Based on [this line in llm-foundry](https://github.com/mosaicml/llm-foundry/blob/e730995c4f0dfdaf9d9d547783739eec48880edb/llmfoundry/utils/builders.py#L311), not sure we ever actually pass question_prelimiter in?

Reply:
well that seems like a bug

Reply:
not sure if you just didn't push this change yet, but i think it is still `prelimiter`?

Reply:
This is `build_icl_dataloader`. llm-foundry calls `get_icl_dataloader` so that's where the true interface is. `get_icl_dataloader` mostly just checks if it needs to run `partition_dataset_by_category` so I'm now considering getting rid of `get_icl_dataloader`. It's only used in this file.

Reply:
oh my b. yeah, down for a refactor there, whatever you think is best

Reply:
Dealing with categorical datasets in this way is convenient, so gonna just leave it as is for now.